modules/library/metadata/src/test/java/org/geotools/measure/UnitFormatterTest.java

```diff
@@ -163,6 +183,13 @@ private static HashMap<Unit<?>, String> getUnitToNameMap(UnitFormat instance) th
         return (HashMap<Unit<?>, String>) unitToNameField.get(instance);
     }
 
+    @SuppressWarnings("unchecked") // reflection in use, cannot be type safe
+    private static UnitFormat getDelegate(BaseUnitFormatter instance) throws Exception {
```

Comment:
This is a problem in Java 17 (should cause a runtime warning). Please try to find a way that does not leverage reflection, as we'll eventually mandate Java 17 as the minim version.

Reply:
removed